builtin/providers/clc/resource_clc_group.go

```diff
@@ -0,0 +1,118 @@
+package clc
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/CenturyLinkCloud/clc-sdk"
+	"github.com/CenturyLinkCloud/clc-sdk/group"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCLCGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCLCGroupCreate,
+		Read:   resourceCLCGroupRead,
+		Update: resourceCLCGroupUpdate,
+		Delete: resourceCLCGroupDelete,
+		Schema: map[string]*schema.Schema{
+			"location_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parent": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"parent_group_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"custom_fields": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
+			},
+		},
+	}
+}
+
+func resourceCLCGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clc.Client)
+	dc := d.Get("location_id").(string)
+	m, err := dcGroups(dc, meta)
+	if err != nil {
+		return fmt.Errorf("Failed pulling groups in location %v - %v", dc, err)
+	}
+	name := d.Get("name").(string)
+	// use an existing group if we have one
+	if m[name] != "" {
+		log.Printf("[INFO] Using EXISTING group: %v => %v", name, m[name])
+		d.SetId(m[name])
+		return nil
+	}
```

Comment:
Part of the intent behind this is the relative inconvenience of working with server groups on clc. In particular, I was trying to allow the user to refer to a group by name rather than the clunkier guid. If instead the code errors on existing group in create, that would more or less require the user to either a) manage groups explicitly or b) book-keep guids. Not sure if the intention comes across through this description. But in the end, if it runs against convention, I'll remove it.